schemas/string_interpolation/definitions.json

```diff
@@ -160,12 +160,11 @@
           "source": {
             "description": "The source of the data.",
             "type": "string",
-            "enum": ["location"]
+            "enum": ["location", "list"]
           },
           "id": {
             "description": "The unique id of the object.",
-            "type": "string",
-            "enum": ["list_item_id", "to_list_item_id"]
+            "type": "string"
```

Comment:
This should restrict the combinations allowed. So only `list_item_id` and `to_list_item_id` are  allowed for `location`, and `first` and `primary_person` for `list`.